### ISSUE TITLE
Code issues regarding implicit differentiation

### ISSUE THREAD

**Issue by JoeyWen99** [xuchen-ethz/snarf#12]
May I ask what the code here means and why you can do this?
https://github.com/xuchen-ethz/snarf/blob/ae0c893cc049f0f8270eaa401e138dff5d4637b9/lib/model/deformer.py#L61
`        correction = xd_opt - xd_opt.detach()
        correction = einsum("bnij,bnj->bni", -grad_inv.detach(), correction)

        # trick for implicit diff with autodiff:
        # xc = xc_opt + 0 and xc' = correction'
        xc = xc_opt + correction`
